File: ProblemSets/6/cash.py
from cs50 import get_float

# The coins
quarter = 25
dime = 10
nickel = 5
penny = 1

# Counter for change owed
counter = 0

# To get the number from user
while True:
    dollars = get_float("Change owed: ")
    if dollars >= 0:
        break

# Converting to cents
cents = int(round(dollars * 100))

# Calculating the result
while cents != 0:

    if cents >= quarter:
        cents -= quarter
        counter += 1
        continue

    elif cents >= dime:
        cents -= dime
        counter += 1
        continue

    elif cents >= nickel:
        cents -= nickel
        counter += 1
        continue

    elif cents >= penny:
        cents -= penny
        counter += 1
        continue

# Printing the result
print(counter)

File: ProblemSets/1/Problem_Set_1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // The variables
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    int counter = 0;

    float dollars;

    // To get the number from user
    while (true)
    {
        dollars = get_float("Change owed: ");
        if (dollars >= 0)
        {
            break;
        }
    }

    // Converting to cents
    int cents = round(dollars * 100);

    // Calculating the result
    while (cents != 0)
    {
        if (cents >= quarter)
        {
            cents -= quarter;
            counter++;
            continue;
        }

        if (cents >= dime)
        {
            cents -= dime;
            counter++;
            continue;
        }

        if (cents >= nickel)
        {
            cents -= nickel;
            counter++;
            continue;
        }

        if (cents >= penny)
        {
            cents -= penny;
            counter++;
            continue;
        }
    }

    // Printing the result
    printf("%i\n", counter);
}